frontend/apps/filemanager/filemanagerhistory.lua

```diff
@@ -171,50 +191,32 @@ function FileManagerHistory:onShowHist()
         _manager = self,
     }
 
+    self.filter = G_reader_settings:readSetting("history_filter", "all")
+    if self.filter ~= "all" then
+        self:getHistStatuses()
+    end
     self:updateItemTable()
     self.hist_menu.close_callback = function()
-        self.statuses_fetched = nil
-        self.filter = nil
         UIManager:close(self.hist_menu)
     end
     UIManager:show(self.hist_menu)
     return true
 end
 
 function FileManagerHistory:showHistDialog()
-    if not self.statuses_fetched then
-        local status
-        for _, v in ipairs(require("readhistory").hist) do
-            if v.dim then
-                status = "deleted"
-            else
-                if DocSettings:hasSidecarFile(v.file) then
-                    local docinfo = DocSettings:open(v.file) -- no io handles created, do not close
-                    if docinfo.data.summary and docinfo.data.summary.status
-                            and docinfo.data.summary.status ~= "" then
-                        status = docinfo.data.summary.status
-                    else
-                        status = "reading"
-                    end
-                else
-                    status = "new"
-                end
-            end
-            v.status = status
-            self.count[status] = self.count[status] + 1
-        end
-        self.statuses_fetched = true
+    if self.filter == "all" then
+        self:getHistStatuses()
```

Comment:
So you will fetch the statuses every time when pressing the left title button in one history session.
Previously it has been done only once in the beginning of the session.

Reply:
Looks like I didn't unweave the spiderweb of calls here properly the first time. There were other issues too. I rolled back and went back to the drawing board (literally). Restored the `statuses_fetched` flag.